assets/js/theme/common/product-details.js

```diff
@@ -591,4 +592,41 @@ export default class ProductDetails extends ProductDetailsBase {
             detail: { productDetails },
         }));
     }
+
+    updateData() {
```

Comment:
as a suggestion here, maybe to give it a more specific name, like `updateDateOption` or `updateDatePickerOption`, `updateDateSelector`?🤔

Reply:
I agree. `updateDateSelector` is suitable in my opinion.